package/samplers/hebo/sampler.py

```diff
@@ -32,14 +97,76 @@ def sample_relative(
             params[name] = params_pd[name].to_numpy()[0]
         return params
 
+    def _sample_relative_stateless(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, float]:
+        if self._constant_liar:
+            target_states = [TrialState.COMPLETE, TrialState.RUNNING]
+        else:
+            target_states = [TrialState.COMPLETE]
+
+        trials = study.get_trials(deepcopy=False, states=target_states)
+        if len([t for t in trials if t.state == TrialState.COMPLETE]) < 1:
+            # note: The backend HEBO implementation use Sobol sampling here.
+            # This sampler does not call `hebo.suggest()` here because
+            # Optuna needs to know search space by running the first trial in Define-by-Run.
+            return {}
+
+        # Assume that the back-end HEBO implementation aims to minimize.
+        if study.direction == StudyDirection.MINIMIZE:
+            worst_values = max([t.values for t in trials if t.state == TrialState.COMPLETE])
+        else:
+            worst_values = min([t.values for t in trials if t.state == TrialState.COMPLETE])
+        sign = 1 if study.direction == StudyDirection.MINIMIZE else -1
+
+        hebo = HEBO(self._convert_to_hebo_design_space(search_space), scramble_seed=self._seed)
+        for t in trials:
+            if t.state == TrialState.COMPLETE:
+                hebo_params = {name: t.params[name] for name in search_space.keys()}
+                hebo.observe(
+                    pd.DataFrame([hebo_params]),
+                    np.asarray([x * sign for x in t.values]),
+                )
+            elif t.state == TrialState.RUNNING:
+                try:
+                    hebo_params = {name: t.params[name] for name in search_space.keys()}
+                except:  # NOQA
+                    # There are one or more params which are not suggested yet.
+                    continue
+                # If `constant_liar == True`, assume that the RUNNING params result in bad values,
+                # thus preventing the simultaneous suggestion of (almost) the same params
+                # during parallel execution.
+                hebo.observe(pd.DataFrame([hebo_params]), np.asarray([worst_values]))
+            else:
+                assert False
+        params_pd = hebo.suggest()
+        params = {}
+        for name in search_space.keys():
+            params[name] = params_pd[name].to_numpy()[0]
+        return params
```

Comment:
```suggestion
        return {name: row[0] for name, row in hebo.suggest().items()}
```

Reply:
Basically agreed. I revised the code to `return {name: row.iloc[0] for name, row in self._hebo.suggest().items() if name in search_space.keys()}`.